<a href="https://colab.research.google.com/github/somas1/CT/blob/main/Pandas/AirBnB_Case_Study_Project_Jupyter_notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment.
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild.

Happy Coding!

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [5]:
air_bnb = pd.read_csv('AB_NYC_2019.csv')
air_bnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park   2787.0   
1  2595                             Skylit Midtown Castle   2845.0   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !   4632.0   
3  3831                   Cozy Entire Floor of Brownstone   4869.0   
4  5022  Entire Apt: Spacious Studio/Loft by central park   7192.0   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room  149.0             1.0                9.0  2018-10-19   
1  Entire home/apt  225.0             1.0               45.0  2019-05-21   
2     Private room  150.0             3.0                0.0         NaN   
3  Entire home/apt   89.0             1.0              270.0  2019-07-05   
4  Entire home/apt   80.0            10.0                9.0  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                             6.0             365.0  
1               0.38                             2.0             355.0  
2                NaN                             1.0             365.0  
3               4.64                             1.0             194.0  
4               0.10                             1.0               0.0

In [231]:
# How many neighborhood groups are available and which shows up the most?
#air_bnb.neighbourhood_group
neighbourhood_groups = air_bnb.groupby(['neighbourhood_group']).size()
neighbourhood_groups

neighbourhood_group
Bronx              501
Brooklyn         12642
Manhattan        13069
Queens            2812
Staten Island      178
dtype: int64

There are five neighborhood groups with Manhattan having the largest number of entries.

In [157]:
# Are private rooms the most popular in manhattan?
Manhattan = air_bnb.query('neighbourhood_group == "Manhattan"')
Manhattan.groupby('room_type').size()

room_type
Entire home/apt    7943
Private room       4884
Shared room         242
dtype: int64

Private rooms are not the most popular rental in Manhattan. Entire home/apartment rentals are the most poplular.

In [210]:
# Which hosts are the busiest and based on their reviews?
reviews = air_bnb.groupby(['number_of_reviews','host_id','host_name']).count().sort_values('number_of_reviews',ascending = False)
reviews.head(10)

id  name  neighbourhood_group  \
number_of_reviews host_id    host_name                                  
629.0             47621202.0 Dona        1     1                    1   
607.0             4734398.0  Jj          1     1                    1   
597.0             4734398.0  Jj          1     1                    1   
594.0             4734398.0  Jj          1     1                    1   
576.0             47621202.0 Dona        1     1                    1   
543.0             37312959.0 Maya        1     1                    1   
540.0             2369681.0  Carol       1     1                    1   
510.0             26432133.0 Danielle    1     1                    1   
488.0             12949460.0 Asa         1     1                    1   
480.0             792159.0   Wanda       1     1                    1   

                                        neighbourhood  latitude  longitude  \
number_of_reviews host_id    host_name                                       
629.0             47621202.0 Dona                   1         1          1   
607.0             4734398.0  Jj                     1         1          1   
597.0             4734398.0  Jj                     1         1          1   
594.0             4734398.0  Jj                     1         1          1   
576.0             47621202.0 Dona                   1         1          1   
543.0             37312959.0 Maya                   1         1          1   
540.0             2369681.0  Carol                  1         1          1   
510.0             26432133.0 Danielle               1         1          1   
488.0             12949460.0 Asa                    1         1          1   
480.0             792159.0   Wanda                  1         1          1   

                                        room_type  price  minimum_nights  \
number_of_reviews host_id    host_name                                     
629.0             47621202.0 Dona               1      1               1   
607.0             4734398.0  Jj                 1      1               1   
597.0             4734398.0  Jj                 1      1               1   
594.0             4734398.0  Jj                 1      1               1   
576.0             47621202.0 Dona               1      1               1   
543.0             37312959.0 Maya               1      1               1   
540.0             2369681.0  Carol              1      1               1   
510.0             26432133.0 Danielle           1      1               1   
488.0             12949460.0 Asa                1      1               1   
480.0             792159.0   Wanda              1      1               1   

                                        last_review  reviews_per_month  \
number_of_reviews host_id    host_name                                   
629.0             47621202.0 Dona                 1                  1   
607.0             4734398.0  Jj                   1                  1   
597.0             4734398.0  Jj                   1                  1   
594.0             4734398.0  Jj                   1                  1   
576.0             47621202.0 Dona                 1                  1   
543.0             37312959.0 Maya                 1                  1   
540.0             2369681.0  Carol                1                  1   
510.0             26432133.0 Danielle             1                  1   
488.0             12949460.0 Asa                  1                  1   
480.0             792159.0   Wanda                1                  1   

                                        calculated_host_listings_count  \
number_of_reviews host_id    host_name                                   
629.0             47621202.0 Dona                                    1   
607.0             4734398.0  Jj                                      1   
597.0             4734398.0  Jj                                      1   
594.0             4734398.0  Jj                       

Dona and Jj have the most reviews.

In [254]:
#Which neighorhood group has the highest average price?
#neighborhood = air_bnb.groupby(['neighbourhood_group'])
#air_bnb['neighbourhood_group', 'price']
air_bnb.groupby(['neighbourhood_group']).mean().sort_values(by = 'price', ascending = False)

<ipython-input-254-69a0a319d9c3>:4: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  air_bnb.groupby(['neighbourhood_group']).mean().sort_values(by = 'price', ascending = False)


id       host_id   latitude  longitude  \
neighbourhood_group                                                     
Manhattan            1.105691e+07  3.263102e+07  40.766408 -73.973628   
Staten Island        1.259631e+07  5.810703e+07  40.613536 -74.103917   
Brooklyn             1.135543e+07  3.082337e+07  40.686242 -73.952785   
Queens               1.292268e+07  5.113167e+07  40.734759 -73.877323   
Bronx                1.337312e+07  5.577123e+07  40.848927 -73.887758   

                          price  minimum_nights  number_of_reviews  \
neighbourhood_group                                                  
Manhattan            183.164511        8.455582          29.283954   
Staten Island        135.820225        3.707865          51.117978   
Brooklyn             125.207008        6.116200          32.679323   
Queens               100.543741        5.237553          43.500356   
Bronx                 89.311377        5.976048          43.477046   

                     reviews_per_month  calculated_host_listings_count  \
neighbourhood_group                                                      
Manhattan                     0.905151                        5.310888   
Staten Island                 1.633355                        2.404494   
Brooklyn                      0.994182                        2.051811   
Queens                        1.519572                        2.339260   
Bronx                         1.424897                        2.229541   

                     availability_365  
neighbourhood_group                    
Manhattan                   93.783763  
Staten Island              224.426966  
Brooklyn                    91.607103  
Queens                     139.478663  
Bronx                      165.435130

Manhattan has the highest average price for rentals.

In [276]:
# Which neighbor hood group has the highest total price?
air_bnb.groupby(['neighbourhood_group']).sum()

<ipython-input-276-2f68f0f31114>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  air_bnb.groupby(['neighbourhood_group']).sum()


id       host_id      latitude     longitude  \
neighbourhood_group                                                           
Bronx                  6699933941  2.794138e+10   20465.31265  -37017.76674   
Brooklyn             143555307845  3.896690e+11  514355.47274 -934911.11023   
Manhattan            144502791314  4.264547e+11  532776.18526 -966761.34496   
Queens                36338589996  1.437823e+11  114546.14327 -207743.03314   
Staten Island          2242142555  1.034305e+10    7229.20932  -13190.49726   

                         price  minimum_nights  number_of_reviews  \
neighbourhood_group                                                 
Bronx                  44745.0          2994.0            21782.0   
Brooklyn             1582867.0         77321.0           413132.0   
Manhattan            2393777.0        110506.0           382712.0   
Queens                282729.0         14728.0           122323.0   
Staten Island          24176.0           660.0             9099.0   

                     reviews_per_month  calculated_host_listings_count  \
neighbourhood_group                                                      
Bronx                           625.53                          1117.0   
Brooklyn                      10648.68                         25939.0   
Manhattan                      9630.81                         69408.0   
Queens                         3689.52                          6578.0   
Staten Island                   253.17                           428.0   

                     availability_365  
neighbourhood_group                    
Bronx                         82883.0  
Brooklyn                    1158097.0  
Manhattan                   1225660.0  
Queens                       392214.0  
Staten Island                 39948.0

Manhattan has the highest total summed price with $2,393,777 made.

In [286]:
#Which top 5 hosts have the highest total price?
top_5 = air_bnb.groupby(['host_id', 'host_name']).sum().sort_values(by = 'price', ascending = False)
top_5.head()

<ipython-input-286-75b3d475310d>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  top_5 = air_bnb.groupby(['host_id', 'host_name']).sum().sort_values(by = 'price', ascending = False)


id    latitude   longitude    price  \
host_id    host_name                                                     
30283594.0 Kara            1343896883  4605.37290 -8360.66980  31354.0   
836168.0   Henry             80246341   448.53355  -813.78842  19500.0   
22541573.0 Ken             1198599504  3300.71897 -5992.72453  17426.0   
16098958.0 Jeremy & Laura   804240348  2894.03334 -5252.61653  14927.0   
1177497.0  Jessica           50161458   447.59267  -813.63698  14850.0   

                           minimum_nights  number_of_reviews  \
host_id    host_name                                           
30283594.0 Kara                    3530.0               65.0   
836168.0   Henry                    330.0              235.0   
22541573.0 Ken                     2430.0               54.0   
16098958.0 Jeremy & Laura          2130.0              133.0   
1177497.0  Jessica                   11.0              595.0   

                           reviews_per_month  calculated_host_listings_count  \
host_id    host_name                                                           
30283594.0 Kara                         3.94                         13673.0   
836168.0   Henry                        3.82                           121.0   
22541573.0 Ken                          4.13                          7047.0   
16098958.0 Jeremy & Laura               5.30                          6816.0   
1177497.0  Jessica                      9.92                           121.0   

                           availability_365  
host_id    host_name                         
30283594.0 Kara                     35269.0  
836168.0   Henry                     3643.0  
22541573.0 Ken                      25349.0  
16098958.0 Jeremy & Laura           21009.0  
1177497.0  Jessica                   3997.0

In [319]:
# Who currently has no (zero) availability with a review count of 100 or more?
air_bnb.query('availability_365 == 0 & number_of_reviews >= 100')

id                                               name  \
8          5203                 Cozy Clean Guest Room - Family Apt   
94        20913                 Charming 1 bed GR8 WBurg LOCATION!   
132       30031                   NYC artists’ loft with roof deck   
174       44221                     Financial District Luxury Loft   
180       45556              Fort Greene, Brooklyn: Center Bedroom   
...         ...                                                ...   
25391  20306217                                   The inner palace   
25430  20332555  Beautiful Apartment near Times Square Central ...   
27007  21381555                                        NYC Dreamer   
27778  21787703      Clean Cozy House less than 10 min from JFK :)   
28042  21901156  Best Deal! Lovely place in Manhattan! Time Squ...   

           host_id   host_name neighbourhood_group       neighbourhood  \
8           7490.0   MaryEllen           Manhattan     Upper West Side   
94         79402.0  Christiana            Brooklyn        Williamsburg   
132       129352.0         Sol            Brooklyn          Greenpoint   
174       193722.0       Coral           Manhattan  Financial District   
180        67778.0        Doug            Brooklyn         Fort Greene   
...            ...         ...                 ...                 ...   
25391  133810925.0      Andrew              Queens            Rosedale   
25430  143944704.0         Ash           Manhattan    Theater District   
27007   50760404.0     Jenifer            Brooklyn       Crown Heights   
27778    7097558.0      Louise              Queens    South Ozone Park   
28042  159156636.0         NaN           Manhattan      Hell's Kitchen   

       latitude  longitude        room_type  price  minimum_nights  \
8      40.80178  -73.96723     Private room   79.0             2.0   
94     40.70984  -73.95775  Entire home/apt  100.0             5.0   
132    40.73494  -73.95030     Private room   50.0             3.0   
174    40.70666  -74.01374  Entire home/apt  196.0             3.0   
180    40.68863  -73.97691     Private room   65.0             2.0   
...         ...        ...              ...    ...             ...   
25391  40.66458  -73.73464     Private room   62.0             1.0   
25430  40.76302  -73.98211  Entire home/apt  239.0             3.0   
27007  40.67178  -73.92362     Private room   80.0             1.0   
27778  40.67078  -73.79164  Entire home/apt   75.0             1.0   
28042  40.75656  -73.99063     Private room  120.0             1.0   

       number_of_reviews last_review  reviews_per_month  \
8                  118.0  2017-07-21               0.99   
94                 168.0  2018-07-22               1.57   
132                193.0  2019-05-20               1.86   
174                114.0  2019-06-20               1.06   
180                206.0  2019-06-30               1.92   
...                  ...         ...                ...   
25391              144.0  2019-06-24               6.21   
25430              104.0  2019-06-22               4.53   
27007              108.0  2019-07-02               5.18   
27778              154.0  2019-06-22               8.24   
28042              109.0  2019-01-01               5.97   

       calculated_host_listings_count  availability_365  
8                                 1.0               0.0  
94                                1.0               0.0  
132                               1.0               0.0  
174                               1.0               0.0  
180                               2.0               0.0  
...                               ...               ...  
25391                             3.0               0.0  
25430                             1.0               0.0  
27007                             2.0               0.0  
27778                             2.0               0.0  
28042                             3.0               0.0  

[157 rows x 16 columns]

In [ ]:
# What host has the highest total of prices and where are they located?



In [ ]:
# When did Danielle from Queens last receive a review?



## Further Questions

1. Which host has the most listings?

2. How many listings have completely open availability?

3. What room_types have the highest review numbers?

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --